lib/handsoap/service.rb

```diff
@@ -265,7 +274,7 @@ def dispatch(doc, action)
         fire_on_log_header { "HandSoap Response [#{dispatch_id}]: length: [#{@http_client.body_str.length}], HTTP-Status: [#{@http_client.response_code}], Content-Type: [#{@http_client.content_type}]" }
         fire_on_log_body { Handsoap.pretty_format_envelope(@http_client.body_str) }
 
-        soap_response = Response.new(@http_client.body_str, envelope_namespace)
+        soap_response = Response.new(@http_client, envelope_namespace)
```

Comment:
I think there's anything serializing access to the HTTP body between the request https://github.com/ManageIQ/handsoap/blob/handsoap_0_2_5/lib/handsoap/service.rb#L263 and accessing the response http://github.com/ManageIQ/handsoap/blob/handsoap_0_2_5/lib/handsoap/service.rb#L268 as it is now for Curl at least.